### ISSUE TITLE
Some types of characters to now appear to be shown in Postico

### ISSUE THREAD

**Issue by geomic** [jakob/Postico#25]
See attached image, for example:
![screenshot 2015-07-01 11 05 36](https://cloud.githubusercontent.com/assets/163444/8452279/6b421096-1fe1-11e5-817c-21662888115a.png)

This is what I get in my Web App, though Postico reports the value as "Monitor access tracks".

**Comment by jakob** [comment]
Could you get the exact contents of the value using `select encode(column_name, 'hex');`?

**Comment by geomic** [comment]
For some reason, I couldn't get encode to work, complains with:

`appdb=# select encode(column_name, 'hex') from table_name;
ERROR:  function encode(character varying, unknown) does not exist
LINE 1: select encode(column_name, 'hex') from table_name;`

But running select name from table gives me:
`Name 1

<U+10FC44><U+10FC19><U+10FC1A><U+10FC13><U+10FC0B><U+10FC17><U+10FC22><U+10FC09><U+10FC1F><U+10FC07><U+10FC0B><U+10FC21><U+10FC1F><U+10FC13><U+10FC0B><U+10FC17><U+10FC22><U+10FC44><U+10FC19><U+10FC1A><U+10FC13>
<U+10FC0B><U+10FC17><U+10FC22><U+10FC09><U+10FC1F><U+10FC07><U+10FC0B><U+10FC21><U+10FC1F><U+10FC13><U+10FC0B><U+10FC17><U+10FC22>Name 2

<U+10FC37><U+10FC19><U+10FC19><U+10FC22><U+10FC13><U+10FC0B><U+10FC17><U+10FC22>Name 3`

So, clearly, erroneous characters here - but can't find a way to see those in Postico (or indeed fix them without running, say, a "update where .. like" query.

**Comment by jakob** [comment]
Okay, so it looks like your text field contains "private use" unicode characters. These can't be displayed because they don't have any encoding. It might be useful to show a placeholder in Postico, but I'm not sure how easy that is to implement.

To fix your issue, you can use the encode function. I forgot that encode only works on "bytea" values, so there is a cast missing. Also, the using 'escape' instead of 'hex' might lead to more readable results. Here's the right code:

    encode(column_name::bytea,'escape')

Then you should get something like this: `\364\217\261\204\364\217\260\231\364\217\260\232Name 2` which you can then fix like this:

    update ... where column_name=E'\364\217\261\204\364\217\260\231\364\217\260\232Name 2'

Note the E for escape string constants. See [Section 4.1.2.2](http://www.postgresql.org/docs/current/static/sql-syntax-lexical.html) of the PostgreSQL manual

**Comment by geomic** [comment]
That's really helpful Jakob, thanks. I did the replacement the messy way (like '%Name 2' etc.) as that table only had a handful of data.

Note that the way the issue came to be in first place, was by pasting data from a semi-locked (apparently...) PDF into a Rails app.

Perhaps � (of Unicode specials) could be used in that case? I think it would be extremely beneficial if these characters weren't "hidden" from Postico.

**Comment by jakob** [comment]
I agree that it would be beneficial to show hidden characters. However, the problem is that adding custom text rendering features to text views (and the table view) is not trivial. Even finding out if a character can be displayed is somewhat complex. I'll keep this in mind, but it'll take me some time to tackle.
